src/bespokelabs/curator/request_processor/openai_batch_request_processor.py

```diff
@@ -743,18 +772,49 @@ async def track_already_submitted_batches(self):
                     else:
                         n_requests = batch_object.request_counts.total
 
-                    if request_file_name in self.tracker.unsubmitted_request_files:
-                        self.tracker.mark_as_submitted(request_file_name, batch_object, n_requests)
-                    else:
-                        # batch objects if not unsubmitted, should be downloaded
-                        assert batch_object.id in self.tracker.downloaded_batches
+                    # For each request file, we only want to keep the latest batch object.
+                    if (
+                        request_file_name not in existing_submitted_batches
```

Comment:
When would there be multiple valid (found, not expired/cancelled) submitted batches?

Reply:
This is very unlikely if we just let curator do its thing but can happen if we are doing some sketchy cache surgery stuff. I'm inclined to keep it just in case we fall into some weird state like that.